### PR TITLE
[PyOV] Update requirements for Python 3.11

### PR DESCRIPTION
**Pull request by p-wysocki**
### Details:
 - Update requirements to versions which support Python 3.11 according to PyPI (wherever possible)

### Tickets:
 - 96724

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

File: tools/mo/requirements_dev.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-coverage>=4.4.2,<=5.5
+coverage>=4.4.2,<=7.0.5
```
[the remark this answers is not in the document]
Comment (ilya-lavrenov) [reply tools/mo/requirements_dev.txt:1]: how to do it? does MO have developer documentation for it?

[the remark this answers is not in the document]
Comment (rkazants) [reply tools/mo/requirements_dev.txt:1]: Yes, it has, I have shared details with @p-wysocki.
Here is small instruction in documentation: https://github.com/openvinotoolkit/openvino/blob/master/tools/mo/README.md

```
How to capture unit-tests coverage
Run tests with:
    coverage run -m unittest discover -p "*_test.py" [-s PATH_TO_DIR]
```

**Review by jiwaszki — APPROVED** [review]
Overall OK, let's check coverage.

File: tools/mo/requirements_dev.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (p-wysocki) [reply tools/mo/requirements_dev.txt:1]: I ran the tests, `coverage==7.0.5` works well. @rkazants

**Review by rkazants — APPROVED** [review]

**Pull request merged**